### ISSUE TITLE
No advective heat transport using coupling of HeatConduction and DeformationFlow

### ISSUE THREAD

**Issue by ThieJan** [ufz/ogs5#162]
Testing the coupling enabled in https://github.com/ufz/ogs5/pull/159 with the Benchmark of P. Vogel from chapter 10.1 in the Benchmark book (Kolditz et al 18) I observed some deviations in the solution especially concerning the temperature profile. Here are the corresponding input file sets for the staggered coupling and the monolithic HM coupled with the T-process:
[T-HM.zip](https://github.com/ufz/ogs5/files/5809562/T-HM.zip)
[T_H_M.ZIP](https://github.com/ufz/ogs5/files/5809563/T_H_M.ZIP)

The H-T-M staggered version is the one used by Peter which fits the analytical solution quite well. Looks like the advective heat transport is not included correctly.

**Comment by ThieJan** [comment]
After some debugging I added the initialization of a coupling pointer for this process coupling combination.
https://github.com/ufz/ogs5/pull/163

However, the results still deviate. Debugging the T-HM example I noticed that in `CFiniteElementStd::CalcAdvection()` the loop runs over 27(=3x3x3) Gaussian points while the data contains valid velocities only for the first 8 points. I suspect this contribites to the deviations.
